tests/src/core/testqgsvectorlayerupdater.cpp

```diff
@@ -0,0 +1,121 @@
+/***************************************************************************
+     testqgsvectorlayerupdater.cpp
+     --------------------------------------
+    Date                 : Feb 2018
+    Copyright            : (C) 2018 by Arnaud Morvan
+    Email                : arnaud dot morvan at coamptocamp dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsvectorlayerupdater.h"
+#include "qgsproject.h"
+#include "qgsvectorlayer.h"
+#include "qgsvectorfilewriter.h"
+#include "qgsapplication.h"
+#include <QObject>
+#include "qgstest.h"
+
+
+class TestQgsVectorLayerUpdater : public QObject
+{
+    Q_OBJECT
+  public:
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+    void init();// will be called before each testfunction is executed.
+    void cleanup();// will be called after every testfunction.
+
+    void update();
+
+  private:
+    QgsVectorLayer mLayer;
+};
+
+void TestQgsVectorLayerUpdater::initTestCase()
+{
+  QgsApplication::init();
+  QgsApplication::initQgis();
+
+  QString testDataDir( TEST_DATA_DIR );  //defined in CMakeLists.txt
+  //QString templatePath = testDataDir + "/polys_with_id.shp";
+  //QString templatePath = testDataDir + "/points.shp";
+  QString tempDir = QDir::tempPath();
```

Comment:
I think you should use a QTemporaryDir here -- otherwise you'll need to check if test_updater.* exists and delete them first, or the copy operations may fail.